static/js/people.js

```diff
@@ -1069,7 +1069,8 @@ export function get_people_for_stream_create() {
     for (const person of active_user_dict.values()) {
         if (!is_my_user_id(person.user_id)) {
             people_minus_you.push({
-                email: person.email,
+                email: get_visible_email(person),
+                show_email: settings_data.show_email(),
```

Comment:
I kinda wonder if it'd be cleaner to just have a single field called `display_email` and have it be either null or a value, rather than having the separate `show_email` field, but that's mostly a readability thing.

Reply:
I tried this but the filter uses email field passed in the `build_person_matcher` code and thus we cannot have email as `null` or `undefined` since this gives error when using `toLowerCase` on it.

Digging into it I feel we should actually change the `build_person_matcher` to match emails only if real emails are available and not wits fake emails since we anyway do not display them. One thing we need to take care of is `build_person_matcher` is used for search-suggestions typeahead also where we still show fake emails as the input box shows it and we need to fix the input box string to have consistency in the typeahead and the actual value in search box after selecting one of the options in typeahead.

Reply:
One thing to keep in mind is that we might need a way to find a user by a unique way to refer to them --and the only way to do that is by their ID.  So I think your proposal is a good idea, but alongside that, I think we need some mechanism to type a user ID and have something good happen.